bootstrap/bootstrap.py

```diff
@@ -90,7 +93,47 @@ def validate_host():
             print("For local development, see http://tljh.jupyter.org/en/latest/contributing/dev-setup.html")
         sys.exit(1)
 
+class LoaderPageRequestHandler(SimpleHTTPRequestHandler):
+    def do_GET(self):
+        if self.path == "/logs":
+            with open("/opt/tljh/installer.log", "rb") as log_file:
+                content = log_file.read()
+                self.wfile.write(content)
+        elif self.path == "/index.html" or self.path == "/favicon.ico":
+            return SimpleHTTPRequestHandler.do_GET(self)
+        elif self.path == "/":
+            self.send_response(302)
+            self.send_header('Location','/index.html')
+            self.end_headers()
+        else:
+            SimpleHTTPRequestHandler.send_error(self, code=403)
+
+def serve_forever(server):
+    try:
+        server.serve_forever()
+    except KeyboardInterrupt:
+        pass
+
 def main():
+    temp_page_flag = "--temporary-page"
+    if temp_page_flag in sys.argv:
+        # Serve the loading page until TLJH builds
+        index_url="https://raw.githubusercontent.com/GeorgianaElena/the-littlest-jupyterhub/in-progress-page/bootstrap/index.html"
+        favicon_url="https://raw.githubusercontent.com/jupyterhub/jupyterhub/master/share/jupyterhub/static/favicon.ico"
+        urllib.request.urlretrieve(index_url, "index.html")
+        urllib.request.urlretrieve(favicon_url, "favicon.ico")
+
+        # If the bootstrap is run to upgrade TLJH, then this will raise an "Address already in use" error
+        try:
+            loading_page_server = HTTPServer(("", 80), LoaderPageRequestHandler)
+            p = multiprocessing.Process(target=serve_forever, args=(loading_page_server,))
+            p.start()
+            # Put the pid after the temp page flag to be passed to the istaller
+            sys.argv.insert(sys.argv.index(temp_page_flag)+1, str(p.pid))
```

Comment:
We should probably make a copy of sys.argv earlier in `main()` and manipulate that, rather than modify sys.argv directly.